lib/modules/platform/gitlab/index.ts

```diff
@@ -1523,3 +1525,15 @@ export async function expandGroupMembers(
   }
   return expandedReviewersOrAssignees;
 }
+
+export function extractRulesFromCodeOwnersFile(
+  cleanedLines: string[],
+): FileOwnerRule[] {
+  const parser = new CodeOwnersParser();
+
+  for (const line of cleanedLines) {
```

Comment:
wouldn't this make more sense to include to the parser directly? so you don't need the weird section state?

Reply:
I ended rewriting the whole class as functions, there is no need for any internal state :slightly_smiling_face: